app/src/main/java/com/example/cactusnotes/validations/LogInValidator.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.cactusnotes.validations
+
+import com.example.cactusnotes.R
+import com.example.cactusnotes.ValidationResult
+
+class LogInValidator : Validator {
+    override fun validate(field: String): ValidationResult {
+        return if (field.isEmpty()) {
```

Comment:
This can look much better if converted to a when block and structured as a single-expression function